### DISCUSSION TITLE
Element wise maximum reduce - similar to numpy.maximum.reduce(arr)

### DISCUSSION THREAD

**Discussion by bruingineer** [rust-ndarray/ndarray#1173] [Q&A]
What would be the best way to make similar functionality to numpy's maximum.reduce feature?
example: 
`np.maximum.reduce([[1, 2, 1, 2],  [3, 0, 3, 0]]`
results in `[3, 2, 3, 2]`

Can I use zip to compare the elements and return the max?

**Comment by jturner314** [comment]
Sure, `Zip` will work if you just want to zip together two rows:

```rust
use ndarray::prelude::*;
use ndarray::Zip;

fn main() {
    let a = array![[1, 2, 1, 2], [3, 0, 3, 0]];

    assert_eq!(
        Zip::from(a.row(0))
            .and(a.row(1))
            .map_collect(|&a, &b| a.max(b)),
        array![3, 2, 3, 2],
    );
}
```

If there may be more than two rows, and you want the maximum element among all the rows, `fold_axis` would work:

```rust
use ndarray::prelude::*;

fn main() {
    let a = array![[1, 2, 1, 2], [3, 0, 3, 0]];

    assert_eq!(
        // Note: if axis 0 has length 0, this will return an array
        // of all `i32::MIN`.
        a.fold_axis(Axis(0), i32::MIN, |&a, &b| a.max(b)),
        array![3, 2, 3, 2],
    );
}
```

**Reply by bruingineer** [reply]
Thank you! I will give that a try
